server/chat/backend/agent/tools/github_fix_tool.py

```
@@ -20,53 +29,624 @@
+                continue
+
+            return [(idx, idx + len(candidate), new)], None
+
+    if not found_any:
+        hints = _find_closest_lines(content, old)
+        if hints:
+            hint_block = "\nClosest lines in the file (copy verbatim — keep quotes, whitespace, indent):\n"
+            for ln, line in hints:
+                hint_block += f"  L{ln}: {line!r}\n"
+            hint_block += (
+                "Pick the right one and include 1–3 surrounding lines as anchor context so "
```

Ruff (0.15.13)

[warning] String contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF001)